dandi/tests/test_metadata.py

```diff
@@ -48,6 +50,29 @@ def test_get_metadata(simple1_nwb: str, simple1_nwb_metadata: Dict[str, Any]) ->
     assert target_metadata == metadata
 
 
+def test_bids_nwb_metadata_integration(bids_examples, tmp_path):
+    """
+    Notes
+    -----
+    * Generating data manually here, as fixture workflow calls `new_dandiset`,
+        which requires spinning up docker:
+        https://github.com/dandi/dandi-cli/pull/1183#discussion_r1061622910
+    """
+
+    source_dpath = os.path.join(bids_examples, "ieeg_epilepsyNWB")
+    dpath = os.path.join(tmp_path, "ieeg_epilepsyNWB")
+    shutil.copytree(source_dpath, dpath)
+
+    file_path = os.path.join(
+        dpath, "sub-01/ses-postimp/ieeg/sub-01_ses-postimp_task-seizure_run-01_ieeg.nwb"
```

Comment:
I don't think this would be legit on windows since would mix `\` and `/` paths... tested on windows -- indeed happens. Even though tests passed (may be some magic is done somewhere down), I would not rely on that I prefer to have it all kosher , smth like

```suggestion
        dpath, *"sub-01/ses-postimp/ieeg/sub-01_ses-postimp_task-seizure_run-01_ieeg.nwb".split("/")
```

to keep it human-readable etc